Sources/NIO/SocketChannel.swift

```diff
@@ -425,13 +431,15 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
         }
         readPending = true
 
-        registerForReadable()
+        if !neverRegistered {
```

Comment:
Are these branches necessary? We have a guard elsewhere.

Reply:
@Lukasa yes I think so as the user may still call `read()` before it is registered via a `ChannelInboundHandler` in the `ServerSocketChannel`s pipeline.

Reply:
these users can be quite inconvenient

Reply:
🤣

Reply:
@normanmaurer Ok, this also needs addressing in a follow-up because it doesn't handle unregistered reads.

Reply:
@Lukasa let me take care.